admin/admin_token_test.go

```diff
@@ -0,0 +1,58 @@
+package admin_test
+
+import (
+	"crypto/ecdsa"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+	"github.com/vocdoni/blind-csp/admin"
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/log"
+)
+
+func TestValidateAdminToken(t *testing.T) {
+	electionIdString := "c5d2460186f7bb73137b620cffde1b3971a0c9023b480c851b70020400000000"
+	var electionId types.HexBytes
+	err := electionId.FromString(electionIdString)
+	assert.NoError(t, err)
+
+	adminToken, err := admin.GenerateAdminToken(electionId)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, adminToken)
+
+	valid, _ := admin.ValidateAdminToken(electionId, adminToken)
+	assert.NoError(t, err)
+	assert.True(t, valid)
+}
+
+func TestValidateSignature(t *testing.T) {
+	// Sample private key
+	privateKey, err := crypto.HexToECDSA("fad9c8855b740a0b7ed4c221dbad0f33a83a49cad6b3fe8d5817ac83d38b6a19")
+	assert.NoError(t, err)
+
+	// Obtain public key
+	publicKey := privateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	if !ok {
+		log.Fatal("error casting public key to ECDSA")
```

Comment:
For tests you should not use log.Fatal() but t.Error() or t.Fatal()

The best approach would be to use the quicktest library (this is the one we use in most of our codes). Since long ago we are migrating all our tests to use this library, which is much more convenient. 

`import	qt "github.com/frankban/quicktest"`

Then for checking an error: `qt.Assert(t, err, qt.IsNil)`

Or a boolean: `qt.Assert(t, ok, qt.IsTrue)`

You can see more examples in vocdoni-node code, ie: https://github.com/vocdoni/vocdoni-node/blob/main/censustree/censustree_test.go